Source/Fuse.Animations/Attract.uno

```diff
@@ -0,0 +1,258 @@
+			void OnValueUpdate3(float3 value) { OnValueUpdate(value); }
+			void OnValueUpdate2(float2 value) { OnValueUpdate(value); }
+			void OnValueUpdate1(float value) { OnValueUpdate(value); }
+			
+			void IListener.OnNewData( IExpression source, object oValue )
+			{
+				var value = float4(0);
+				int size = 0;
+				if (!TryMarshalToAnimFloat4(oValue, out value, out size))
+				{
+					//invalid values can simply discard the simulation (forcing a new one to be created when
+					//a good value arrives)
+					CleanSimulation();
+					return;
+				}
+
+				NeedSim(size);
+				
+				if (_simulation1 != null)
+					_simulation1.SetValue( value.X, OnValueUpdate1 );
+				if (_simulation2 != null)
+					_simulation2.SetValue( value.XY, OnValueUpdate2 );
+				if (_simulation3 != null)
+					_simulation3.SetValue( value.XYZ, OnValueUpdate3 );
+				if (_simulation4 != null)
+					_simulation4.SetValue( value, OnValueUpdate4 );
+			}
+			
+			void NeedSim(int size)
+			{
+				if (size != _simSize)
+				{
+					CleanSimulation();
+					_simSize = size;
+				}
+				
+				if (size < 0 || size > 4)
+				{
+					Fuse.Diagnostics.InternalError( "Unexpected size for attract: " + size, this );
+					return;
+				}
+					
+				if (size == 1 && _simulation1 == null)
+				{
+					_simulation1 = new DestinationBehavior<float>();
+					_simulation1.Motion = _attract._config;
+				}
+				else if (size == 2 && _simulation2 == null)
+				{
+					_simulation2 =new DestinationBehavior<float2>();
+					_simulation2.Motion = _attract._config;
+				}
+				else if (size == 3 && _simulation3 == null)
+				{
+					_simulation3 =new DestinationBehavior<float3>();
+					_simulation3.Motion = _attract._config;
+				}
+				else if (size == 4 && _simulation4 == null)
+				{
+					_simulation4 =new DestinationBehavior<float4>();
+					_simulation4.Motion = _attract._config;
+				}
+			}
+			
+			void CleanSimulation()
+			{
+				if (_simulation4 != null)
+				{
+					_simulation4.Unroot();
+					_simulation4 = null;
+				}
+				if (_simulation3 != null)
+				{
+					_simulation3.Unroot();
+					_simulation3 = null;
+				}
+				if (_simulation2 != null)
+				{
+					_simulation2.Unroot();
+					_simulation2 = null;
+				}
+				if (_simulation1 != null)
+				{
+					_simulation1.Unroot();
+					_simulation1 = null;
+				}
+			}
+			
+			public void Dispose()
+			{
+				if (_sourceSub != null)
+				{
+					_sourceSub.Dispose();
+					_sourceSub = null;
+				}
+				_target = null;
+				_attract = null;
+				CleanSimulation();
+			}
+		}
+		
+		/**
+			Converts a value to a float4 suitable for animation, where unspecified values are 0.
+			
+			NOTE: Perhaps this should be made a standard method in Marshal, there must be more places that
+				need this logic. The base logic was just copied from Marshal.ToFloat4 (modified to add 0's) -- Size handling was removed until we can build the special logic needed to support it's animation.
```

Comment:
I'll do one better and remove the entire function. It's part of Fuse.Marshal now.